### ISSUE TITLE
THEY JUST STOLE EVERYONE'S RECOVERY PARTITION AND PREVENTED ALL CUSTOM RECOVERY OPTIONS.

### ISSUE THREAD

**Issue by firepacket** [TeamWin/android_device_oneplus_guacamole#1]
You are on the right track with /data/reserve/reserve.img in patchfakepartitions.sh.
I got screwed by OnePlus who literally stole my recovery partition from me without my permission by upgrading me EXPLICITLY AGAINST MY WILL while I was running Pie then I found myself forced into running some Chinese OxygenOS and unable to boot recovery partition or boot recovery on boot partition. PLEASE HELP.

I want to go back to Pie normal. But I can't, unless you guys can fix this. Flashing the recovery to boot is freezing my phone just like trying to boot from it, and flashing the install file is sending my phone into a boot loop where I can see TWRP appear for a split second but then it reboots.

So something is a little off. Please keep working on this.

This is disgusting behavior from a company who promised it's customers freedom. They went back on their promise.
They need to sued. This is illegal.

**Comment by ipoopedmypantsuups** [comment]
This is disgusting behavior from what I assumed to be a reputable company. For shame.

**Comment by f355** [comment]
`fastboot boot twrp.img` does not work for you two for some reason?

this is pathetic. nobody is trying to limit your freedum - you can unlock the bootloader and do whatever you want, including but not limited to writing your own OS from scratch if you're so inclined, with any partitions you desire. the company has never promised to cater to your wishes for the partition layout in their stock firmware.

**Comment by firepacket** [comment]
> `fastboot boot twrp.img` does not work for you two for some reason?

No. It absolutely does not.

You can do stuff on your OP7Pro
### THAT I CANNOT DO ON MINE

> this is pathetic. nobody is trying to limit your freedum

### WRONG

**___You DO NOT know what you're talking about___,**
Different versions of the same phone have been made. Apparently it matters where you buy these from. There are Chinese-controlled knockoffs which appear to have different hardware AND software. They have their own separate version they sell to unsuspecting customers on Amazon.


###  `fastboot boot twrp.img` does not work on our OP7Pros

**Comment by f355** [comment]
@firepacket oh wow. this amount of markup will get you banned in a whim if you use it in a more active repository/conversation, just sayin'.

on a more serious note, how can anyone help you if you've bought a knock-off with hardware different from 7pro/guacamole? it is a different phone, Oneplus didn't sell it to you, you yourself went to a shady dealer to get it, because it was much cheaper. how is that a problem of Oneplus and/or TWRP?

also, this is your freedom in a nutshell lol - buy whatever from the free market for whatever amount you wish to spend, and bear the consequences if it's not the thing that you expected.

**Comment by firepacket** [comment]
> @firepacket oh wow. this amount of markup will get you banned in a whim if you use it in a more active repository/conversation, just sayin'.
> 
> on a more serious note, how can anyone help you if you've bought a knock-off with hardware different from 7pro/guacamole? it is a different phone, Oneplus didn't sell it to you, you yourself went to a shady dealer to get it, because it was much cheaper. how is that a problem of Oneplus and/or TWRP?
> 
> also, this is your freedom in a nutshell lol - buy whatever from the free market for whatever amount you wish to spend, and bear the consequences if it's not the thing that you expected.

It's the only OnePlus 7 Pro I've been able to find available since I heard about them. It is a REAL OnePlus7Pro, and I eventually got it working the way I want it. But it **is different** than yours and if you did a little research you would know there are different versions.

I did not know it would be any different, it's not like they advertised it. It's called a scam, smart-alack. It's *your* freedom in a nutshell too, so don't whine next time someone scams you.

**Comment by f355** [comment]
> It's called a scam, smart-alack.

of course it is a scam, but is it oneplus who scammed you? did you buy it from oneplus themselves? maybe, from an official oneplus reseller? no, you've bought it from some random dude on amazon marketplace. go file a warranty claim with that dude, what can I say.

**Comment by firepacket** [comment]
Oneplus wasn't selling it anywhere officially. Nobody was. And it wasn't cheaper. The Amazon seller had good reviews (for all you smug people who think Amazon is great LOL)

I already told you, I figured out how to defeat their spyware and am now happy with the phone. But there are still things it can't do. Like fastboot boot *.img

**Comment by f355** [comment]
Amazon is shit, sure :) and good that you have figured it out, I'm happy for you!

**Comment by firepacket** [comment]
> > It's called a scam, smart-alack.
> 
> of course it is a scam, but is it oneplus who scammed you? did you buy it from oneplus themselves? maybe, from an official oneplus reseller? no, you've bought it from some random dude on amazon marketplace. go file a warranty claim with that dude, what can I say.

There was no place to buy this phone I waited as long as I could without having a phone and even these were expensive with 2 in stock. Amazon seller reviews looked fine.

It obviously came from their manufactures in China, so yea, OnePlus shares some of the blame for choosing sleazy manufacturers. I'm sure Apple has ways to prevent against this.

**Comment by f355** [comment]
not really. most phone manufacturers - Oneplus and Apple included - have their versions of phones for the Chinese market due to more strict regulatory and more lax QA requirements, they're typically cheaper and aren't supposed to be sold elsewhere, but people still buy and resell them for profit.

Apple is making two physical SIM iPhones not available outside China, a bunch of my Russian friends have them.
Fluke, the famous American manufacturer of test equipment that's as American as apple pie and has had government contracts for the past 50 years, makes stripped-down, quarter-of-the-price versions of their multimeters for the Chinese market.

your phone was never meant to be sold outside China. it's not a knock-off, it's just what it is. again, I'm happy that you made it work but there's no fault by Oneplus here.

**Comment by f355** [comment]
what's your model number btw? GM1910?

**Comment by firepacket** [comment]
1915

**Comment by firepacket** [comment]
After researching, I figured it was an unlocked, then heavily modified T-Mobile edition.

**Comment by f355** [comment]
right, that makes perfect sense.
